unison-core/src/Unison/NameSegment.hs

```diff
@@ -0,0 +1,41 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Unison.NameSegment where
+
+import Unison.Prelude
+
+import qualified Data.Text                     as Text
+import qualified Unison.Hashable               as H
+
+-- Represents the parts of a name between the `.`s
+newtype NameSegment = NameSegment { toText :: Text } deriving (Eq, Ord)
+
+-- Split text into segments. A smarter version of `Text.splitOn` that handles
+-- the name `.` properly.
+segments' :: Text -> [Text]
```

Comment:
What does the `'` mean here?

Reply:
What kinds of `Text` things should we be passing to `segments'`?  How does it relate to `NameSegment`?

Reply:
Would it make sense to be an implementation detail in `Name`?

Reply:
The `'` is here just since `segments` is taken by the `Name` module and these two are usually used in the same context. `NameSegment.segments'` is the internal implementation of `Name.segments`.

`segments'` will accept absolutely any text, just like `Text.splitOn`. It's purely a replacement for `Text.splitOn '.'`, as per the comment.

Reply:
`segments'` has to be called in places where we can't import `Name` because of a dumb circular dependency. It actually is an implementation detail, but one we have to expose. It's just "a better `Text.splitOn '.'`"

Reply:
Oh, ok. :-\  Would you add a note to the source to that effect?

Reply:
Or can you pull some stuff out to a new module, to avoid the circular dependency?